edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.Http.Test/TwinsControllerTest.cs

```diff
@@ -163,6 +163,190 @@ public async Task TestInvokeMethodOnModuleWithResponsePayload()
+            Assert.Equal(new JRaw(responsePayload), methodResult.Payload);
+            long methodResultByteCount = Encoding.UTF8.GetByteCount(JsonConvert.SerializeObject(methodResult));
+            Assert.Equal(methodResultByteCount, testController.Response.ContentLength);
+            Assert.Equal(objectResult.StatusCode, (int)HttpStatusCode.OK);
+        }
+
+        [Fact]
+        public async Task TestInvokeMethodOnModuleWithResponsePayloadWithUmlautAndResponseContentLengthHeaderIsEqualToByteCount()
+        {
+            var identity = Mock.Of<IIdentity>(i => i.Id == "edgedevice/module1");
+            ActionExecutingContext actionExecutingContext = this.GetActionExecutingContextMock(identity);
+
+            string responsePayload = "{ \"resp1\" : \"respvalüe1\" }";
+            var directMethodResponse = new DirectMethodResponse(Guid.NewGuid().ToString(), Encoding.UTF8.GetBytes(responsePayload), 200);
+            var edgeHub = new Mock<IEdgeHub>();
+            edgeHub.Setup(e => e.InvokeMethodAsync(It.Is<string>(i => i == identity.Id), It.IsAny<DirectMethodRequest>()))
+                .ReturnsAsync(directMethodResponse);
+
+            var validator = new Mock<IValidator<MethodRequest>>();
+            validator.Setup(v => v.Validate(It.IsAny<MethodRequest>()));
+
+            var httpContext = new DefaultHttpContext(); // or mock a `HttpContext`
+            var controllerContext = new ControllerContext()
+            {
+                HttpContext = httpContext,
+            };
+            var testController = new TwinsController(Task.FromResult(edgeHub.Object), validator.Object)
+            {
+                ControllerContext = controllerContext
+            };
+            testController.OnActionExecuting(actionExecutingContext);
+
+            string toDeviceId = "edgedevice";
+            string toModuleId = "module2";
+            string command = "showdown";
+            string payload = "{ \"prop1\" : \"value1\" }";
+
+            var methodRequest = new MethodRequest(command, new JRaw(payload));
+            IActionResult actionResult = await testController.InvokeModuleMethodAsync(WebUtility.UrlEncode(toDeviceId), WebUtility.UrlEncode(toModuleId), methodRequest);
+
+            Assert.NotNull(actionResult);
+            var objectResult = actionResult as ObjectResult;
+            Assert.NotNull(objectResult);
+            var methodResult = objectResult.Value as MethodResult;
+            Assert.NotNull(methodResult);
+            Assert.Equal(200, methodResult.Status);
+            Assert.Equal(new JRaw(responsePayload), methodResult.Payload);
+            Assert.Equal(new JRaw(responsePayload), methodResult.Payload);
+            long methodResultByteCount = Encoding.UTF8.GetByteCount(JsonConvert.SerializeObject(methodResult));
+            Assert.Equal(methodResultByteCount, testController.Response.ContentLength);
+            Assert.Equal(objectResult.StatusCode, (int)HttpStatusCode.OK);
```

Comment:
Shouldn't expected be first argument?

Reply:
The test should check if the `Content-Length` header (actual) has the same byte count as the text being sent (expected).

Reply:
SebastianRoether's comment is for line 347, where the parameters are indeed backwards.

Reply:
You are right. I will edit them in the other tests as well.